example/relocatable_function/main.cpp

```diff
@@ -0,0 +1,49 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+
+#include <iostream>
+
+KOKKOS_RELOCATABLE_FUNCTION void count_even(const long i, long& lcount);
+
+int main() {
+  Kokkos::ScopeGuard scope_guard;
```

Comment:
In examples we really should showcase passing argc and argv through.
Also now that you removed the early exit, it makes little sense to use ScopeGuard